src/libsystemd-network/sd-dhcp-lease.c

```diff
@@ -579,6 +595,17 @@ int dhcp_lease_parse_options(uint8_t code, uint8_t len, const uint8_t *option,
                         return r;
 
                 break;
+
+        case DHCP_OPTION_VENDOR_SPECIFIC:
+                if (len >= 1) {
+                        free(lease->vendor_specific);
+                        lease->vendor_specific = memdup(option, len);
+                        if (!lease->vendor_specific)
+                                return -errno;
```

Comment:
`memdup()` is our own function, and we don't mess with errno. You have to explicitly return `-ENOMEM' here.